Evolutionary_loop/Loop_Parts/Part_B/part_b_job1.py

```diff
@@ -0,0 +1,229 @@
+        with open ( xf_proj/ 'Simulations/.nextSimulationNumber',"w+") as f:
+            f.write( str(gen*npop + 1) )
+# NECESSARY ? NECESSARY ? NECESSARY ? NECESSARY ? NECESSARY ? NECESSARY ? 
+
+
+## INITIALIZE XMACRO
+
+    # list of 60 frequencies used in XFdtd VPol simulations.
+    freq_list = [83.33, 100.00, 116.67, 133.33, 150.00, 166.67, 183.34, 200.00,
+        216.67, 233.34, 250.00, 266.67, 283.34, 300.00, 316.67, 333.34, 350.00,
+        366.67, 383.34, 400.01, 416.67, 433.34, 450.01, 466.67, 483.34, 500.01,
+        516.68, 533.34, 550.01, 566.68, 583.34, 600.01, 616.68, 633.34, 650.01,
+        666.68, 683.35, 700.01, 716.68, 733.35, 750.01, 766.68, 783.35, 800.01,
+        816.68, 833.35, 850.02, 866.68, 883.35, 900.02, 916.68, 933.35, 950.02,
+        966.68, 983.35, 1000.00, 1016.70, 1033.40, 1050.00, 1066.70] 
+
+    sim_path = xmacros_dir/ 'simulation_PEC.xmacro'
+    # If simulation.xmacro exists, empty it (without changing permission)
+    sp.run(f'[ -f {sim_path} ] && > {sim_path}', shell=True)
+    
+    # Begin writing some run-specific variables to the xmacro
+    with open (sim_path, "w+") as f:
+        f.write(f'var NPOP = {npop};\n'
+                f'var indiv = {indiv};\n'
+                f'//Factor of {geo_factor} frequency\n'
+                'var freq = [')
+        f.write(', '.join( "%.2f"%(freq*geo_factor) for freq in freq_list) )
+        f.write('] \n')
+    
+    if (gen == 0 and indiv == 1):
+        with open (sim_path,"a") as f:
+            f.write('if(indiv==1){\n'
+                    f'App.saveCurrentProjectAs(\"{working_dir}/Run_Outputs/'
+                    f'{run_name}/{run_name}\");\n'
+                    '}\n')
+
+    os.umask(0)                # umask(0) needed to ensure chmod works properly.
+    os.chmod(sim_path, 0o775 ) # This is the same as Bash chmod 775 ${sim_path}.
+
+
+## CAT SECTION
+
+    # The rest of simulation_PEC.xmacro is built from two skeleton text files.
+    if curved == 0:                            
+        if nsections == 1:                     # straight-side symmetric bicone
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton_GPU.txt'
+                   f'>> {sim_path}', shell=True)
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton2_GPU.txt'
+                   f'>> {sim_path}', shell=True)
+
+        else:                                  # straight-side asymmetric bicone
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton_Sep.txt'
+                   f'>> {sim_path}', shell=True)
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton2_Sep.txt'
+                   f'>> {sim_path}', shell=True)
+    else:                                      # curved-side bicone
+        sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton_curved.txt'
+               f'>> {sim_path}', shell=True)
+        sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton2_curved.txt'
+               f'>> {sim_path}', shell=True)
```

Comment:
```suggestion
    if curved == 0:                            
        if nsections == 1:                     # straight-side symmetric bicone
            fst = 'simulationPECmacroskeleton_GPU.txt'
            snd = 'simulationPECmacroskeleton2_GPU.txt'
        else:                                  # straight-side asymmetric bicone
            fst = 'simulationPECmacroskeleton_Sep.txt'
            snd = 'simulationPECmacroskeleton2_Sep.txt'
    else:                                      # curved-side bicone
        fst = 'simulationPECmacroskeleton_curved.txt'
        snd = 'simulationPECmacroskeleton2_curved.txt'

    with open(sim_path, 'w') as outfile:
        outfile.write((xmacros_dir / fst).read_text())
        outfile.write((xmacros_dir / snd).read_text())
```
This should be equivalent

Reply:
should the mode be `a` instead of `w`?